### ISSUE TITLE
Output is being clipped(?) on larger file saves.

### ISSUE THREAD

**Issue by Basketcasesoftware** [SixLabors/ImageSharp#1769]
### Prerequisites

- [x ] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of ImageSharp
- [ ] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
I've been trying to modify an existing png image of 2500x2500 pixels but I doubt the image format matters. When I try to save out the the new image it does save as 2500x2500 pixels (png and jpg formats tested), but only the first 61 rows are filled with valid data; the rest is filler (i.e. transparency in the case of png files). 

### Steps to Reproduce
```C#
using (Image<Rgba32> SourceOverlay = Image.Load<Rgba32>(@"C:\Users\Richard's Tower\source\repos\ResplitTiles\Images\WorldMaps\Overlay.png"))
{
    using (Image<Rgba32> TargetOverlay = new(SourceOverlay.Width, SourceOverlay.Height))
    {
        for (int WorldY = 0; WorldY < SourceOverlay.Height; WorldY++)
            for (int WorldX = 0; WorldX < SourceOverlay.Width; WorldX++)
            {
                TargetOverlay[WorldX, WorldY] = SourceOverlay[WorldX, WorldY];
            }

        TargetOverlay.SaveAsPng(@"C:\Users\Richard's Tower\source\repos\ResplitTiles\Images\WorldMaps\OverlayEnlarged.jpg");
    }
}
```

### System Configuration
- ImageSharp version: 1.0.3
- Other ImageSharp packages and versions: None
- Environment (Operating system, version and so on): Windows 10 Pro. OS Build 19043.1263
- .NET Framework version: .Net 6.0- 
- Additional information:  Edit: I'm using x64 Debug mode for testing.

**Comment by brianpopow** [comment]
@Basketcasesoftware does this really also happen with jpg or only with png? If its only png this might be fixed already with #1704.

Can you try 1.0.4 in that case? We have made this release especially for that case.

**Comment by Basketcasesoftware** [comment]
Update: I just did a test of the pixel read process. The problem isn't in the file saves, but the file loads. At pixel {X: 0, Y: 61} the read value is {R: 0, G: 0, B: 0, A: 0}. Clearly not what is expected!

**Comment by tocsoft** [comment]
> Update: I just did a test of the pixel read process. The problem isn't in the file saves, but the file loads. At pixel {X: 0, Y: 61} the read value is {R: 0, G: 0, B: 0, A: 0}. Clearly not what is expected!

did you try updating to version 1.0.4 of ImageSharp? version 1.0.3 has a known issue loading PNGs on .net 6 which 1.0.4 resolved.

**Comment by JimBobSquarePants** [comment]
@Basketcasesoftware How did you get on?

**Comment by JimBobSquarePants** [comment]
I'm going to close this as I believe the issue has already been solved by #1707
